idea/src/org/jetbrains/kotlin/idea/inspections/SetterBackingFieldAssignmentInspection.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.*
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.findDescendantOfType
+
+class SetterBackingFieldAssignmentInspection : AbstractKotlinInspection(), CleanupLocalInspectionTool {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean, session: LocalInspectionToolSession): PsiElementVisitor {
+        return propertyAccessorVisitor(fun(accessor) {
+            if (!accessor.isSetter) return
+            if (accessor.hasLowerVisibilityThanProperty()) return
+            val bodyExpression = accessor.bodyExpression as? KtBlockExpression ?: return
+            if (bodyExpression.findDescendantOfType<KtBinaryExpression> {
+                    it.operationToken == KtTokens.EQ && it.left?.text == "field"
```

Comment:
What do you think about other assignment operators, like `PLUSEQ`? I'm not sure exactly but think that with them we should report this.